### PR TITLE
refactor: remove unneeded lazy initialization

### PR DESCRIPTION
**Pull request by matklad**
Ratio::new_raw exists specifically for the purposes of const evaluation.
It's a separate function because it can't check that denominator is not
zero due to the limitations of the rust type system.

But we are using it in a const context with literal values here, so we
don't need the check anyway.

### CODE REVIEW

**Review by bowenwang1996 — APPROVED** [review]
Epic!

**Comment by matklad** [comment]
Note that the tests are failing though, and  I have no idea why. I pushed the branch several times, and this sanity check failed every time

**Comment by ailisp** [comment]
sanity test fails on this:
> python3 scripts/state/update_res.py check
neard/res/genesis_config.json does not match `near init` generated
Please update by run scripts/state/update_res.py

Although it's weird why this PR changes what `near init` generated `~/.near/genesis.json become different

**Comment by bowenwang1996** [comment]
> Although it's weird why this PR changes what near init generated `~/.near/genesis.json become different

Could it be that the serialization format for `Rational` has changed?

**Comment by ailisp** [comment]
> Could it be that the serialization format for Rational has changed?

But other PR's check pass

**Comment by ailisp** [comment]
I rerun update_res.py and got this diff, but his doesn't look like anything changed:
```
--- a/neard/res/genesis_config.json
+++ b/neard/res/genesis_config.json
@@ -6,234 +6,234 @@
   "num_block_producer_seats": 50,
   "num_block_producer_seats_per_shard": [
     50
-  ], 
+  ],
   "avg_hidden_validator_seats_per_shard": [
     0
-  ], 
-  "dynamic_resharding": false, 
+  ],
+  "dynamic_resharding": false,
   "protocol_upgrade_stake_threshold": [
-    4, 
-    5
-  ], 
-  "protocol_upgrade_num_epochs": 2, 
-  "epoch_length": 500, 
-  "gas_limit": 1000000000000000, 
-  "min_gas_price": "1000000000", 
-  "max_gas_price": "10000000000000000000000", 
-  "block_producer_kickout_threshold": 90, 
-  "chunk_producer_kickout_threshold": 90, 
+    8,
+    10
+  ],
+  "protocol_upgrade_num_epochs": 2,
+  "epoch_length": 500,
+  "gas_limit": 1000000000000000,
+  "min_gas_price": "1000000000",
+  "max_gas_price": "10000000000000000000000",
+  "block_producer_kickout_threshold": 90,
+  "chunk_producer_kickout_threshold": 90,
   "online_min_threshold": [
-    9, 
+    9,
     10
-  ], 
+  ],
   "online_max_threshold": [
-    99, 
+    99,
     100
-  ], 
+  ],
   "gas_price_adjustment_rate": [
-    1, 
+    1,
     100
-  ], 
+  ],
   "runtime_config": {
-    "storage_amount_per_byte": "90900000000000000000", 
+    "storage_amount_per_byte": "90900000000000000000",
     "transaction_costs": {
       "action_receipt_creation_config": {
-        "send_sir": 108059500000, 
-        "send_not_sir": 108059500000, 
+        "send_sir": 108059500000,
+        "send_not_sir": 108059500000,
         "execution": 108059500000
-      }, 
+      },
       "data_receipt_creation_config": {
         "base_cost": {
-          "send_sir": 4697339419375, 
-          "send_not_sir": 4697339419375, 
+          "send_sir": 4697339419375,
+          "send_not_sir": 4697339419375,
           "execution": 4697339419375
-        }, 
+        },
         "cost_per_byte": {
-          "send_sir": 59357464, 
-          "send_not_sir": 59357464, 
+          "send_sir": 59357464,
+          "send_not_sir": 59357464,
           "execution": 59357464
         }
-      }, 
+      },
       "action_creation_config": {
         "create_account_cost": {
-          "send_sir": 99607375000, 
-          "send_not_sir": 99607375000, 
+          "send_sir": 99607375000,
+          "send_not_sir": 99607375000,
           "execution": 99607375000
-        }, 
+        },
         "deploy_contract_cost": {
-          "send_sir": 184765750000, 
-          "send_not_sir": 184765750000, 
+          "send_sir": 184765750000,
+          "send_not_sir": 184765750000,
           "execution": 184765750000
-        }, 
+        },
         "deploy_contract_cost_per_byte": {
-          "send_sir": 6812999, 
-          "send_not_sir": 6812999, 
+          "send_sir": 6812999,
+          "send_not_sir": 6812999,
           "execution": 6812999
-        }, 
+        },
         "function_call_cost": {
-          "send_sir": 2319861500000, 
-          "send_not_sir": 2319861500000, 
+          "send_sir": 2319861500000,
+          "send_not_sir": 2319861500000,
           "execution": 2319861500000
-        }, 
+        },
         "function_call_cost_per_byte": {
-          "send_sir": 2235934, 
-          "send_not_sir": 2235934, 
+          "send_sir": 2235934,
+          "send_not_sir": 2235934,
           "execution": 2235934
-        }, 
+        },
         "transfer_cost": {
-          "send_sir": 115123062500, 
-          "send_not_sir": 115123062500, 
+          "send_sir": 115123062500,
+          "send_not_sir": 115123062500,
           "execution": 115123062500
-        }, 
+        },
         "stake_cost": {
-          "send_sir": 141715687500, 
-          "send_not_sir": 141715687500, 
+          "send_sir": 141715687500,
+          "send_not_sir": 141715687500,
           "execution": 102217625000
-        }, 
+        },
         "add_key_cost": {
           "full_access_cost": {
-            "send_sir": 101765125000, 
-            "send_not_sir": 101765125000, 
+            "send_sir": 101765125000,
+            "send_not_sir": 101765125000,
             "execution": 101765125000
-          }, 
+          },
           "function_call_cost": {
-            "send_sir": 102217625000, 
-            "send_not_sir": 102217625000, 
+            "send_sir": 102217625000,
+            "send_not_sir": 102217625000,
             "execution": 102217625000
-          }, 
+          },
           "function_call_cost_per_byte": {
-            "send_sir": 1925331, 
-            "send_not_sir": 1925331, 
+            "send_sir": 1925331,
+            "send_not_sir": 1925331,
             "execution": 1925331
           }
-        }, 
+        },
         "delete_key_cost": {
-          "send_sir": 94946625000, 
-          "send_not_sir": 94946625000, 
+          "send_sir": 94946625000,
+          "send_not_sir": 94946625000,
           "execution": 94946625000
-        }, 
+        },
         "delete_account_cost": {
-          "send_sir": 147489000000, 
-          "send_not_sir": 147489000000, 
+          "send_sir": 147489000000,
+          "send_not_sir": 147489000000,
           "execution": 147489000000
         }
-      }, 
+      },
       "storage_usage_config": {
-        "num_bytes_account": 100, 
+        "num_bytes_account": 100,
         "num_extra_bytes_record": 40
-      }, 
+      },
       "burnt_gas_reward": [
-        3, 
+        3,
         10
-      ], 
+      ],
       "pessimistic_gas_price_inflation_ratio": [
-        103, 
+        103,
         100
       ]
-    }, 
+    },
     "wasm_config": {
       "ext_costs": {
-        "base": 264768111, 
-        "contract_compile_base": 35445963, 
-        "contract_compile_bytes": 216750, 
-        "read_memory_base": 2609863200, 
-        "read_memory_byte": 3801333, 
-        "write_memory_base": 2803794861, 
-        "write_memory_byte": 2723772, 
-        "read_register_base": 2517165186, 
-        "read_register_byte": 98562, 
-        "write_register_base": 2865522486, 
-        "write_register_byte": 3801564, 
-        "utf8_decoding_base": 3111779061, 
-        "utf8_decoding_byte": 291580479, 
-        "utf16_decoding_base": 3543313050, 
-        "utf16_decoding_byte": 163577493, 
-        "sha256_base": 4540970250, 
-        "sha256_byte": 24117351, 
-        "keccak256_base": 5879491275, 
-        "keccak256_byte": 21471105, 
-        "keccak512_base": 5811388236, 
-        "keccak512_byte": 36649701, 
-        "log_base": 3543313050, 
-        "log_byte": 13198791, 
-        "storage_write_base": 64196736000, 
-        "storage_write_key_byte": 70482867, 
-        "storage_write_value_byte": 31018539, 
-        "storage_write_evicted_byte": 32117307, 
-        "storage_read_base": 56356845750, 
-        "storage_read_key_byte": 30952533, 
-        "storage_read_value_byte": 5611005, 
-        "storage_remove_base": 53473030500, 
-        "storage_remove_key_byte": 38220384, 
-        "storage_remove_ret_value_byte": 11531556, 
-        "storage_has_key_base": 54039896625, 
-        "storage_has_key_byte": 30790845, 
-        "storage_iter_create_prefix_base": 0, 
-        "storage_iter_create_prefix_byte": 0, 
-        "storage_iter_create_range_base": 0, 
-        "storage_iter_create_from_byte": 0, 
-        "storage_iter_create_to_byte": 0, 
-        "storage_iter_next_base": 0, 
-        "storage_iter_next_key_byte": 0, 
-        "storage_iter_next_value_byte": 0, 
-        "touching_trie_node": 16101955926, 
-        "promise_and_base": 1465013400, 
-        "promise_and_per_promise": 5452176, 
-        "promise_return": 560152386, 
-        "validator_stake_base": 911834726400, 
+        "base": 264768111,
+        "contract_compile_base": 35445963,
+        "contract_compile_bytes": 216750,
+        "read_memory_base": 2609863200,
+        "read_memory_byte": 3801333,
+        "write_memory_base": 2803794861,
+        "write_memory_byte": 2723772,
+        "read_register_base": 2517165186,
+        "read_register_byte": 98562,
+        "write_register_base": 2865522486,
+        "write_register_byte": 3801564,
+        "utf8_decoding_base": 3111779061,
+        "utf8_decoding_byte": 291580479,
+        "utf16_decoding_base": 3543313050,
+        "utf16_decoding_byte": 163577493,
+        "sha256_base": 4540970250,
+        "sha256_byte": 24117351,
+        "keccak256_base": 5879491275,
+        "keccak256_byte": 21471105,
+        "keccak512_base": 5811388236,
+        "keccak512_byte": 36649701,
+        "log_base": 3543313050,
+        "log_byte": 13198791,
+        "storage_write_base": 64196736000,
+        "storage_write_key_byte": 70482867,
+        "storage_write_value_byte": 31018539,
+        "storage_write_evicted_byte": 32117307,
+        "storage_read_base": 56356845750,
+        "storage_read_key_byte": 30952533,
+        "storage_read_value_byte": 5611005,
+        "storage_remove_base": 53473030500,
+        "storage_remove_key_byte": 38220384,
+        "storage_remove_ret_value_byte": 11531556,
+        "storage_has_key_base": 54039896625,
+        "storage_has_key_byte": 30790845,
+        "storage_iter_create_prefix_base": 0,
+        "storage_iter_create_prefix_byte": 0,
+        "storage_iter_create_range_base": 0,
+        "storage_iter_create_from_byte": 0,
+        "storage_iter_create_to_byte": 0,
+        "storage_iter_next_base": 0,
+        "storage_iter_next_key_byte": 0,
+        "storage_iter_next_value_byte": 0,
+        "touching_trie_node": 16101955926,
+        "promise_and_base": 1465013400,
+        "promise_and_per_promise": 5452176,
+        "promise_return": 560152386,
+        "validator_stake_base": 911834726400,
         "validator_total_stake_base": 911834726400
-      }, 
-      "grow_mem_cost": 1, 
-      "regular_op_cost": 3856371, 
+      },
+      "grow_mem_cost": 1,
+      "regular_op_cost": 3856371,
       "limit_config": {
-        "max_gas_burnt": 200000000000000, 
-        "max_gas_burnt_view": 200000000000000, 
-        "max_stack_height": 16384, 
-        "initial_memory_pages": 1024, 
-        "max_memory_pages": 2048, 
-        "registers_memory_limit": 1073741824, 
-        "max_register_size": 104857600, 
-        "max_number_registers": 100, 
-        "max_number_logs": 100, 
-        "max_total_log_length": 16384, 
-        "max_total_prepaid_gas": 300000000000000, 
-        "max_actions_per_receipt": 100, 
-        "max_number_bytes_method_names": 2000, 
-        "max_length_method_name": 256, 
-        "max_arguments_length": 4194304, 
-        "max_length_returned_data": 4194304, 
-        "max_contract_size": 4194304, 
-        "max_length_storage_key": 4194304, 
-        "max_length_storage_value": 4194304, 
-        "max_promises_per_function_call_action": 1024, 
+        "max_gas_burnt": 200000000000000,
+        "max_gas_burnt_view": 200000000000000,
+        "max_stack_height": 16384,
+        "initial_memory_pages": 1024,
+        "max_memory_pages": 2048,
+        "registers_memory_limit": 1073741824,
+        "max_register_size": 104857600,
+        "max_number_registers": 100,
+        "max_number_logs": 100,
+        "max_total_log_length": 16384,
+        "max_total_prepaid_gas": 300000000000000,
+        "max_actions_per_receipt": 100,
+        "max_number_bytes_method_names": 2000,
+        "max_length_method_name": 256,
+        "max_arguments_length": 4194304,
+        "max_length_returned_data": 4194304,
+        "max_contract_size": 4194304,
+        "max_length_storage_key": 4194304,
+        "max_length_storage_value": 4194304,
+        "max_promises_per_function_call_action": 1024,
         "max_number_input_data_dependencies": 128
       }
-    }, 
+    },
     "account_creation_config": {
-      "min_allowed_top_level_account_length": 0, 
+      "min_allowed_top_level_account_length": 0,
       "registrar_account_id": "registrar"
     }
-  }, 
+  },
   "validators": [
     {
-      "account_id": "test.near", 
-      "public_key": "ed25519:9BmAFNRTa5mRRXpSAm6MxSEeqRASDGNh2FuuwZ4gyxTw", 
+      "account_id": "test.near",
+      "public_key": "ed25519:9BmAFNRTa5mRRXpSAm6MxSEeqRASDGNh2FuuwZ4gyxTw",
       "amount": "50000000000000000000000000000000"
     }
-  ], 
-  "transaction_validity_period": 100, 
+  ],
+  "transaction_validity_period": 100,
   "protocol_reward_rate": [
-    1, 
+    1,
     10
-  ], 
+  ],
   "max_inflation_rate": [
-    1, 
-    20
-  ], 
-  "total_supply": "2050000000000000000000000000000000", 
-  "num_blocks_per_year": 31536000, 
-  "protocol_treasury_account": "test.near", 
-  "fishermen_threshold": "10000000000000000000000000", 
-  "minimum_stake_divisor": 10, 
+    5,
+    100
+  ],
+  "total_supply": "2050000000000000000000000000000000",
+  "num_blocks_per_year": 31536000,
+  "protocol_treasury_account": "test.near",
+  "fishermen_threshold": "10000000000000000000000000",
+  "minimum_stake_divisor": 10,
   "records": []
-}
+}
```

File: neard/src/config.rs [hunk 1]
```diff
@@ -140,21 +139,19 @@ pub const VALIDATOR_KEY_FILE: &str = "validator_key.json";
...
-    /// Maximum inflation rate per year
-    pub static ref MAX_INFLATION_RATE: Rational = Rational::new(5, 100);
+/// Maximum inflation rate per year
+pub const MAX_INFLATION_RATE: Rational = Rational::new_raw(5, 100);
 
-    /// Protocol upgrade stake threshold.
-    pub static ref PROTOCOL_UPGRADE_STAKE_THRESHOLD: Rational = Rational::new(8, 10);
-}
+/// Protocol upgrade stake threshold.
+pub const PROTOCOL_UPGRADE_STAKE_THRESHOLD: Rational = Rational::new_raw(8, 10);
```
Comment (ailisp) [inline neard/src/config.rs:154]: ah so this should be changed to `Rational::new_raw(4, 5)`, because `Rational::new` does normalization it to 4,5

File: neard/src/config.rs [hunk 2]
```diff
@@ -140,21 +139,19 @@ pub const VALIDATOR_KEY_FILE: &str = "validator_key.json";
...
 
-    /// Protocol treasury reward
-    pub static ref PROTOCOL_REWARD_RATE: Rational = Rational::new(1, 10);
+/// Protocol treasury reward
+pub const PROTOCOL_REWARD_RATE: Rational = Rational::new_raw(1, 10);
 
-    /// Maximum inflation rate per year
-    pub static ref MAX_INFLATION_RATE: Rational = Rational::new(5, 100);
+/// Maximum inflation rate per year
+pub const MAX_INFLATION_RATE: Rational = Rational::new_raw(5, 100);
```
Comment (ailisp) [inline neard/src/config.rs:151]: also this `1,20`

**Review by frol — APPROVED** [review]

**Review by ailisp — APPROVED** [review]
